scarb/src/process.rs

```diff
@@ -87,9 +87,21 @@ mod imp {
     }
 }
 
-/// Runs the process, waiting for completion, and mapping non-success exit codes to an error.
 #[tracing::instrument(level = "trace", skip_all)]
 pub fn exec(cmd: &mut Command, config: &Config) -> Result<()> {
+    fn print_fn(line: String, _config: &Config) {
+        debug!("{line}");
+    }
+    exec_piping(cmd, config, print_fn)
+}
+
+/// Runs the process, waiting for completion, and mapping non-success exit codes to an error.
+#[tracing::instrument(level = "trace", skip_all)]
+pub fn exec_piping(
+    cmd: &mut Command,
+    config: &Config,
+    print_fn: fn(String, &Config),
```

Comment:
why not?

```suggestion
    print_fn: impl Fn(&str),
```

1. This would allow you to pass closures, so no `Config` plumbing
2. Passing `&str` would reduce allocation